src/libostree/ostree-repo-pull.c

```diff
@@ -1538,7 +1538,7 @@ scan_one_metadata_object_c (OtPullData         *pull_data,
       /* Even though we already have the commit, we always try to (re)fetch the
        * detached metadata before scanning it, in case new signatures appear.
        * https://github.com/projectatomic/rpm-ostree/issues/630 */
-      if (!g_hash_table_contains (pull_data->fetched_detached_metadata, tmp_checksum))
+      if (!pull_data->remote_repo_local && !g_hash_table_contains (pull_data->fetched_detached_metadata, tmp_checksum))
```

Comment:
Good idea! Though it needed a conditional for the object type (and a strdup).  Done below :arrow_down: